
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Read Mapping &#8212; sphinx_test_1 0.0.1 documentation</title>
    <link rel="stylesheet" href="../_static/classic.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="MetaQUAST" href="metaquast.html" />
    <link rel="prev" title="Assembly Evaluation" href="index.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="metaquast.html" title="MetaQUAST"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Assembly Evaluation"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">sphinx_test_1 0.0.1 documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="index.html" accesskey="U">Assembly Evaluation</a> &#187;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="read-mapping">
<h1>Read Mapping<a class="headerlink" href="#read-mapping" title="Permalink to this headline">¶</a></h1>
<p>In this part of the tutorial we will look at the assemblies by mapping the reads to the assembled contigs.
Different tools exists for mapping reads to genomic sequences such as <a class="reference external" href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">bowtie</a> or <a class="reference external" href="http://bio-bwa.sourceforge.net/">bwa</a>. Today, we will use the tool BBMap.</p>
<p>BBMap: Short read aligner for DNA and RNA-seq data. Capable of handling arbitrarily large genomes with millions of scaffolds. Handles Illumina, PacBio, 454, and other reads; very high sensitivity and tolerant of errors and numerous large indels. Very fast. See the <a class="reference external" href="http://sourceforge.net/projects/bbmap/">BBMap home page</a> for more info.</p>
<p><code class="docutils literal notranslate"><span class="pre">bbmap</span></code> needs to build an index for the contigs sequences before it can map the reads onto them. Here is an example command line for mapping the reads back to the MEGAHIT assembly:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">workdir</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">megahit_out</span>

<span class="n">qsub</span> <span class="o">-</span><span class="n">cwd</span> <span class="o">-</span><span class="n">N</span> <span class="n">bbmap_index</span> <span class="o">-</span><span class="n">l</span> <span class="n">mtc</span><span class="o">=</span><span class="mi">1</span> <span class="o">-</span><span class="n">b</span> <span class="n">y</span> \
<span class="o">/</span><span class="n">vol</span><span class="o">/</span><span class="n">cmg</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">bbmap</span><span class="o">.</span><span class="n">sh</span> <span class="n">ref</span><span class="o">=</span><span class="n">final</span><span class="o">.</span><span class="n">contigs</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>Now that we have an index, we can map the reads:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qsub</span> <span class="o">-</span><span class="n">cwd</span> <span class="o">-</span><span class="n">pe</span> <span class="n">multislot</span> <span class="mi">24</span> <span class="o">-</span><span class="n">N</span> <span class="n">bbmap</span> <span class="o">-</span><span class="n">l</span> <span class="n">mtc</span><span class="o">=</span><span class="mi">1</span> <span class="o">-</span><span class="n">b</span> <span class="n">y</span> \
<span class="o">/</span><span class="n">vol</span><span class="o">/</span><span class="n">cmg</span><span class="o">/</span><span class="nb">bin</span><span class="o">/</span><span class="n">bbmap</span><span class="o">.</span><span class="n">sh</span> <span class="ow">in</span><span class="o">=../</span><span class="n">read1</span><span class="o">.</span><span class="n">fq</span> <span class="n">in2</span><span class="o">=../</span><span class="n">read2</span><span class="o">.</span><span class="n">fq</span> <span class="n">out</span><span class="o">=</span><span class="n">megahit</span><span class="o">.</span><span class="n">sam</span> <span class="n">bamscript</span><span class="o">=</span><span class="n">sam2bam</span><span class="o">.</span><span class="n">sh</span> <span class="n">threads</span><span class="o">=</span><span class="mi">24</span>
</pre></div>
</div>
<p><code class="docutils literal notranslate"><span class="pre">bbmap</span></code> produces output in <a class="reference external" href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM format</a> by default, usually you want to convert this into a sorted BAM file. <code class="docutils literal notranslate"><span class="pre">bbmap</span></code> creates a shell script which can be used to convert <code class="docutils literal notranslate"><span class="pre">bbmap</span></code>’s output into BAM format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qsub</span> <span class="o">-</span><span class="n">cwd</span> <span class="o">-</span><span class="n">pe</span> <span class="n">multislot</span> <span class="mi">4</span> <span class="o">-</span><span class="n">N</span> <span class="n">bbmap_sam2bam</span> <span class="o">-</span><span class="n">l</span> <span class="n">mtc</span><span class="o">=</span><span class="mi">1</span> <span class="n">sam2bam</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>SAM and BAM files can be viewed and manipulated with <a class="reference external" href="http://samtools.sourceforge.net/">SAMtools</a>. Let’s first build an index for the FASTA file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">faidx</span> <span class="n">final</span><span class="o">.</span><span class="n">contigs</span><span class="o">.</span><span class="n">fa</span>
</pre></div>
</div>
<p>To look at the BAM file use:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">samtools</span> <span class="n">view</span> <span class="n">megahit_sorted</span><span class="o">.</span><span class="n">bam</span> <span class="o">|</span> <span class="n">less</span>
</pre></div>
</div>
<p>We will use <a class="reference external" href="http://www.broadinstitute.org/igv/">IGV: Integrative Genomics Viewer</a>
to look at the mappings:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="o">~/</span><span class="n">workdir</span><span class="o">/</span><span class="n">assembly</span><span class="o">/</span><span class="n">megahit_out</span>
<span class="n">igv</span><span class="o">.</span><span class="n">sh</span>
</pre></div>
</div>
<p>Now let’s look at the mapped reads:</p>
<ol class="arabic simple">
<li>Load the contig sequences into IGV. Use the menu <code class="docutils literal notranslate"><span class="pre">Genomes-&gt;Load</span> <span class="pre">Genome</span> <span class="pre">from</span> <span class="pre">File...</span></code></li>
<li>Load the BAM file into IGV. Use menu <code class="docutils literal notranslate"><span class="pre">File-&gt;Load</span> <span class="pre">from</span> <span class="pre">File...</span></code></li>
<li>Load the predicted genes as another track. Use menu <code class="docutils literal notranslate"><span class="pre">File-&gt;Load</span> <span class="pre">from</span> <span class="pre">File...</span></code> to load the GFF file.</li>
</ol>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="index.html"
                        title="previous chapter">Assembly Evaluation</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="metaquast.html"
                        title="next chapter">MetaQUAST</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="../_sources/day3/mapping.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="metaquast.html" title="MetaQUAST"
             >next</a> |</li>
        <li class="right" >
          <a href="index.html" title="Assembly Evaluation"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="../index.html">sphinx_test_1 0.0.1 documentation</a> &#187;</li>
          <li class="nav-item nav-item-1"><a href="index.html" >Assembly Evaluation</a> &#187;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, Ulas Karaoz.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.7.5.
    </div>
  </body>
</html>